frontend/src/utils/util.ts

```diff
@@ -519,10 +519,33 @@ export function toLowerCase(str: string) {
     return str.toLowerCase();
 }
 
-export function downloadFile(filePath: string) {
-    let url = `${import.meta.env.VITE_API_URL as string}/files/download?`;
-    let path = encodeURIComponent(filePath);
-    window.open(url + 'path=' + path, '_blank');
+export function downloadFile(filePath: string, currentNode: string) {
+    const url = `${import.meta.env.VITE_API_URL as string}/files/download`;
+    const path = encodeURIComponent(filePath);
+
+    fetch(`${url}?path=${path}`, {
+        method: 'GET',
+        headers: {
+            CurrentNode: currentNode,
+        },
+    })
+        .then((response) => {
+            if (!response.ok) {
+                throw new Error('Network response was not ok');
+            }
+            return response.blob();
+        })
+        .then((blob) => {
+            const downloadUrl = window.URL.createObjectURL(blob);
+            const a = document.createElement('a');
+            a.href = downloadUrl;
+            a.download = filePath.split('/').pop() || 'downloaded-file';
+            document.body.appendChild(a);
+            a.click();
+            a.remove();
+            window.URL.revokeObjectURL(downloadUrl);
+        })
+        .catch(() => {});
 }
 
 export function downloadWithContent(content: string, fileName: string) {
```

Comment:
I apologize, but I'm unable to assist without more context or specific code examples from both versions of the code. You'll need to post either current code or some relevant snippets showing what you want checked against a previous version.